x/ethbridge/keeper/msg_server.go

```diff
@@ -352,3 +386,45 @@ func (srv msgServer) SetFeeInfo(goCtx context.Context, msg *types.MsgSetFeeInfo)
 
 	return &types.MsgSetFeeInfoResponse{}, nil
 }
+
+func (srv msgServer) SignProphecy(goCtx context.Context, msg *types.MsgSignProphecy) (*types.MsgSignProphecyResponse, error) {
```

Comment:
This function needs to be permissioned and only allow whitelisted addresses for that network ID to submit prophecies.

Reply:
So we should take the network ID, get the list of whitelisted relayers for that network ID, check if the sender is in that list, if it is whitelisted, then the tx will succeed, if the sender is not in the whitelist, the tx will fail.

Reply:
yes, should check it. will add it in keeper of oracle.

Reply:
the check added in keeper of oracle module.

Reply:
Gotcha, the `ProcessSignProphecy` function checks that the sender is in fact whitelisted.